modules/stochastic_tools/doc/content/modules/stochastic_tools/install_pytorch.md

```diff
@@ -0,0 +1,164 @@
+Next, we configure MOOSE with libtorch. This can be achieved by simply calling the
+configure script the following way:
+
+```bash
+./configure --with-libtorch
+```
+
+!alert! note
+If you would like to use a custom libtorch distribution or a manually compiled
+version, the destination of the libtorch directory can be supplied to the
+configure script by
+
+```bash
+./configure --with-libtorch=/path/to/my/custom/libtorch
+```
+
+!alert-end!
+
+The last step is to compile MOOSE stochastic tools with libtorch support:
+
+```bash
+cd modules/stochastic_tools
+make -j 8
+```
+
+The same procedure is expected to work with every other module and the framework itself.
+
+
+## Installation on a local Linux workstation
+
+The difference compared to Mac machines is that the moose conda packages (compiler stack) have been
+built using `libc` v2.12 meaning that it cannot be used to set up the framework with
+libtorch. For this reason, we need to install PETSc and Libmesh manually, assuming that
+the default compiler stack on the system uses libc 2.27+ (most modern distributions,
+such as Ubuntu actually do have suitable compilers). For this first, clone the
+repository, then use the corresponding scripts as:
+
+```bash
+git clone https://github.com/idaholab/moose.git
+cd moose
+./scripts/update_and_rebuild_petsc.sh
+./scripts/update_and_rebuild_libmesh.sh
+```
+
+See [this link](https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html)
+for more information on the scripts used for setting up `libmesh` and `PETSc`.
+From this point on, the procedure is more or less the same as the installation
+on a Mac machine. We use the same scripts to setup a suitable
+precompiled version of `libtorch`:
+
+```bash
+./scripts/setup_libtorch.sh
+./configure --with-libtorch
+```
+
+which creates a `framework/contrib/libtorch` directory. This folder
+will contain the necessary header files and shared object files which can be used
+for dynamic linking.
+
+Now, we can move on compile MOOSE with libtorch.
+
+```bash
+cd modules/stochastic_tools
+make -j 8
+```
+
+## Installation on a HPC machine
+
+The installation process is very similar to the one on a local Linux machine.
+However, in some cases, we may have to build our own compiler stack/mpicc.
+For this, follow the instructions
+[here](https://mooseframework.inl.gov/getting_started/installation/manual_installation_gcc.html).
+Once this is done and we ensured that our compilers were built using glibc 2.7+,
+we can proceed to clone moose, build `petsc` and `libmesh`, download `libtorch` and
+built moose with `libtorch`:
+
+```bash
+git clone https://github.com/idaholab/moose.git
+cd moose
+./scripts/update_and_rebuild_petsc.sh
+./scripts/update_and_rebuild_libmesh.sh
+./scripts/setup_libtorch.sh
+./configure --with-libtorch
+cd modules/stochastic_tools
+make -j 8
+```
+
+## Testing the installation
+
+Basic neural network trainer and surrogate classes have been implemented in the stochastic tools
```

Comment:
This is the only info that should be in stochastic_tools. With a link to the framework torch install.

Reply:
Addressed!